### DISCUSSION TITLE
Why does the installation of my R package 'interpolation' fail on Mac?

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7383] [Q&A] [answered]
Hello,

My R package **interpolation** is successfully installed on numerous systems (even with Valgrind), but the installation fails on Mac.

In view of [this temporary compilation report](https://www.r-project.org/nosvn/R.check/r-release-macos-x86_64/interpolation-00install.html), would you know whether this is due to a too old version of **mpfr** maybe? There were some errors with an outdated version of **mpfr** but I'm not sure there are the same.

**Comment by mglisse** [comment]
Is the version of clang available somewhere? It looks like https://bugs.llvm.org/show_bug.cgi?id=43266 , but that was supposedly fixed a while ago.

**Reply by stla** [reply]
Apple LLVM version 10.0.0 (clang-1000.10.44.4); GNU Fortran (GCC) 8.2.0

**Reply by mglisse** [reply, marked as answer]
Apparently that means LLVM-6, while the highly similar bug report mentions a fix in LLVM-9 (probably Xcode 11.4). If there is a way to use a more recent compiler, that would be best. Otherwise, you could try to patch GMP with
```diff
-#define mpz_sgn(Z) ((Z)->_mp_size < 0 ? -1 : (Z)->_mp_size > 0) 
+#define mpz_sgn(Z) (((Z)->_mp_size < 0) ? -1 : ((Z)->_mp_size > 0)) 
```
(and similar for the next 2 lines) to see if that helps work around this compiler bug.

**Reply by stla** [reply]
Thanks!
